code/modules/shuttle/custom_shuttle.dm

```diff
@@ -0,0 +1,229 @@
+	dat += "<A href='?src=[REF(user)];mach_close=computer'>Close</a>"
+
+	popup = new(user, "computer", M ? M.name : "shuttle", 350, 450)
+	popup.set_content("<center>[dat]</center>")
+	popup.set_title_image(usr.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/custom_shuttle/Topic(href, href_list)
+	if(..())
+		return
+	usr.set_machine(src)
+	src.add_fingerprint(usr)
+	if(!allowed(usr))
+		to_chat(usr, "<span class='danger'>Access denied.</span>")
+		return
+
+	if(href_list["calculate"])
+		calculateStats()
+		ui_interact(usr)
+		return
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return
+	if(M.launch_status == ENDGAME_LAUNCHED)
+		return
+	if(href_list["setloc"])
+		SetTargetLocation(href_list["setloc"])
+		ui_interact(usr)
+		return
+	else if(href_list["fly"])
+		Fly()
+		ui_interact(usr)
+		return
+
+/obj/machinery/computer/custom_shuttle/proc/calculateDistance(var/obj/docking_port/stationary/port)
+	var/deltaX = port.x - x
+	var/deltaY = port.y - y
+	var/deltaZ = (port.z - z) * Z_DIST
+	return sqrt(deltaX * deltaX + deltaY * deltaY + deltaZ * deltaZ)
+
+/obj/machinery/computer/custom_shuttle/proc/linkShuttle(var/new_id)
+	shuttleId = new_id
+	possible_destinations = "shuttle[new_id]_custom"
+
+/obj/machinery/computer/custom_shuttle/proc/calculateStats(var/useFuel = FALSE, var/dist = 0)
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return FALSE
+	//Reset data
+	calculated_mass = 0
+	calculated_dforce = 0
+	calculated_speed = 0
+	calculated_engine_count = 0
+	calculated_consumption = 0
+	calculated_cooldown = 0
+	calculated_fuel_less_thrusters = 0
+	calculated_non_operational_thrusters = 0
+	//Calculate all the data
+	var/list/areas = M.shuttle_areas
+	for(var/shuttleArea in areas)
+		for(var/each in shuttleArea)
+			var/atom/atom = each
+			if(!atom)
+				continue
+			calculated_mass ++
+			if(!istype(atom, /obj/machinery/shuttle/engine))
+				continue
+			var/obj/machinery/shuttle/engine/E = atom
+			E.check_setup(FALSE)
+			if(!E.thruster_active)	//Skipover thrusters with no valid heater
+				calculated_non_operational_thrusters ++
+				continue
+			if(!E.attached_heater.hasFuel(dist * E.fuel_use) && useFuel)
+				calculated_fuel_less_thrusters ++
+				continue
+			calculated_engine_count++
+			calculated_dforce += E.thrust
+			calculated_consumption += E.fuel_use
+			calculated_cooldown = max(calculated_cooldown, E.cooldown)
+	//This should really be accelleration, but its a 2d spessman game so who cares
+	if(calculated_mass == 0)
+		return FALSE
+	calculated_speed = (calculated_dforce*1000) / (calculated_mass*100)
+	return TRUE
+
+/obj/machinery/computer/custom_shuttle/proc/consumeFuel(var/dist)
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return FALSE
+	//Calculate all the data
+	var/list/areas = M.shuttle_areas
+	//fix this shitcode plox
+	for(var/shuttleArea in areas)
+		for(var/each in shuttleArea)
```

Comment:
I know there is a better method for this, will have to codedive to find something else

Reply:
This has to be like this as it is calculating mass too :(

Reply:
Fixed 🙂